src/App.tsx

```diff
@@ -1,27 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+enum FUNCS {
+  GET_ALL = 'getAll',
+  GET_FIRST_5 = 'getFirstFive',
+  GET_RED = 'getRed',
+}
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [functionGetData, setFunctionGetData] = useState(FUNCS.GET_ALL);
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  useEffect(() => {
+    if (functionGetData === FUNCS.GET_RED) {
+      getRedGoods()
+        .then(goodsFromServer => {
+          setGoods(goodsFromServer);
+        });
+    } else if (functionGetData === FUNCS.GET_FIRST_5) {
+      get5First()
+        .then(goodsFromServer => {
+          setGoods(goodsFromServer);
+        });
+    } else {
+      getAll()
+        .then(goodsFromServer => {
+          setGoods(goodsFromServer);
+        });
+    }
+  }, [functionGetData]);
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => setFunctionGetData(FUNCS.GET_ALL)}
```

Comment:
You can create some function for setting goods and pass the callback there. Then just call the callback inside this function and set the goods to the goods state. The logic is too complicated now